distribution/xfer/download.go

```diff
@@ -70,7 +86,7 @@ type DownloadDescriptor interface {
 	// before).
 	DiffID() (layer.DiffID, error)
 	// Download is called to perform the download.
-	Download(ctx context.Context, progressOutput progress.Output) (io.ReadCloser, int64, error)
+	Download(ctx context.Context, progressOutput progress.Output, cacheDir string) (io.ReadCloser, int64, error)
```

Comment:
@tiborvass was chatting with @stevvooe and he wasn't sure what you're meaning; could you explain a bit more? :blush: